python_modules/dagster-graphql/dagster_graphql/schema/backfill.py

```diff
@@ -115,6 +117,52 @@ class Meta:
         name = "BulkActionStatus"
 
 
+# TODO rebase on top of https://github.com/dagster-io/dagster/pull/17244 once it merges, to reuse
+# this object type
+class GrapheneAssetBackfillTargetPartitions(graphene.ObjectType):
+    class Meta:
+        name = "AssetBackfillTargetPartitions"
+
+    ranges = graphene.List(
+        graphene.NonNull("dagster_graphql.schema.partition_sets.GraphenePartitionKeyRange")
+    )
+    partitionKeys = graphene.List(graphene.NonNull(graphene.String))
+
+    def __init__(self, partition_subset: PartitionsSubset):
+        from dagster_graphql.schema.partition_sets import GraphenePartitionKeyRange
+
+        if isinstance(partition_subset, TimeWindowPartitionsSubset):
+            ranges = [
+                GraphenePartitionKeyRange(start, end)
+                for start, end in partition_subset.get_partition_key_ranges()
+            ]
+            partition_keys = None
+        else:  # Default partitions subset
+            ranges = None
+            partition_keys = partition_subset.get_partition_keys()
+
+        super().__init__(
+            ranges=ranges,
+            partitionKeys=partition_keys,
+        )
+
+
+class GrapheneAssetPartitions(graphene.ObjectType):
```

Comment:
The type on the `partitions` field implies that this class is specific to backfills, but the class doesn't have backfill in its name. I think that is worth remedying, either by making `GrapheneAssetBackfillTargetPartitions` not backfill-specific or by making this class backfill-specific.

Reply:
Yeah I want to rename `GrapheneAssetBackfillTargetPartitions` to `GraphenePartitionSelection` or something like that

Reply:
I would prefer to do that in a separate PR though, since Ben's changes in https://github.com/dagster-io/dagster/pull/17244 actually added the `GrapheneAssetBackfillTargetPartitions` type

(I should have just stacked my PR on top)